apps/verticalhome/js/app.js

```diff
@@ -158,6 +193,179 @@
 
     },
 
+    saveAllPinnedApps: function(){
+
+      var pinnedItems= app.pinnedAppsManager.items;
+      if(this.deletedApps.length>0){
+        for(var x = 0; x < pinnedItems.length; x++){
+          if(pinnedItems[x].element.classList.contains('hidden-item')){
+            pinnedItems.splice(x,1);
+          }
+        }
+
+        for(var z = 0; z < this.deletedApps.length; z++){
+          app.itemStore.deletePinnedAppItem(this.deletedApps[z]);
+        }
+        this.deletedApps.splice(0,this.deletedApps.length);
+      }
+
+      for(var i = 0; i<pinnedItems.length; i++){
+        var obj = {};
+
+        if(pinnedItems[i].entry && pinnedItems[i].entry!=null){
+
+          var curEntry = pinnedItems[i].entry;
+          obj.name = curEntry.name;
+
+          obj.manifestURL = curEntry.manifestURL ||
+            curEntry.targetApp.manifestURL;
+
+          obj.icon = curEntry.icon || curEntry.icon.baseURI;
+
+          obj.index = pinnedItems[i].index;
+
+          app.itemStore.savePinnedAppItem(obj);
+
+        }
```

Comment:
It seems me that first and third cycle in this function could be as one cycle. 
Because condition of this cycles are equals. 
Could you rewrite this in more optimized form?